File: static/CWE-327/csharp/patched.cs
// Remediation for CWE-327: Use of a Broken or Risky Cryptographic Algorithm
// Fix: Replace MD5/DES with SHA-256 hashing and AES encryption using securely generated keys.

using System.Security.Cryptography;
using System.Text;

namespace GuineaPig.Sast
{
    class Patched_327
    {
        public byte[] HashAndEncrypt(string userInput)
        {
            byte[] data = Encoding.UTF8.GetBytes(userInput);

            using (var sha256 = SHA256.Create())
            {
                byte[] digest = sha256.ComputeHash(data);

                using (var aes = Aes.Create())
                {
                    aes.GenerateKey();
                    aes.GenerateIV();
                    using (var encryptor = aes.CreateEncryptor())
                    {
                        return encryptor.TransformFinalBlock(digest, 0, digest.Length);
                    }
                }
            }
        }
    }
}

File: static/CWE-327/java/patched.java
// Remediation for CWE-327: Use of a Broken or Risky Cryptographic Algorithm
// Fix: Replace DES/MD5 with AES/GCM for encryption and SHA-256 for hashing.

import java.security.MessageDigest;
import javax.crypto.Cipher;
import javax.crypto.spec.GCMParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.security.SecureRandom;

class Patched_327 {
    public byte[] strongCrypto(String userInput, byte[] key) throws Exception {
        Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
        byte[] iv = new byte[12];
        new SecureRandom().nextBytes(iv);
        cipher.init(Cipher.ENCRYPT_MODE,
            new SecretKeySpec(key, "AES"), new GCMParameterSpec(128, iv));
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        return md.digest(userInput.getBytes("UTF-8"));
    }
}
